compiler/record-minmax/src/RecordMinMax.cpp

```diff
@@ -39,6 +39,35 @@ using DataType = luci_interpreter::DataType;
 namespace
 {
 
+uint32_t numElements(const luci::CircleNode *node)
+{
+  uint32_t num_elements = 1;
+  for (uint32_t i = 0; i < node->rank(); i++)
+    num_elements *= node->dim(i).value();
+
+  assert(num_elements > 0); // FIX_CALLER_UNLESS
+
+  return num_elements;
+}
+
+// Throw exception if input has one of the following conditions.
+// 1. Have unknown dimension
+// 2. Number of elements is 0
+void checkInputDimension(const luci::CircleInput *input)
+{
+  uint32_t num_elements = 1;
+  for (uint32_t i = 0; i < input->rank(); i++)
+  {
+    if (!input->dim(i).known())
+      throw std::runtime_error(input->name() + " has unknown dimension");
+
+    num_elements *= input->dim(i).value();
+  }
+
+  if (num_elements == 0)
+    throw std::runtime_error(input->name() + " is a zero-sized input");
```

Comment:
I think calculating the number of elements is duplicated :)
How about following?

```suggestion
  for (uint32_t i = 0; i < input->rank(); i++)
    if (!input->dim(i).known())
      throw std::runtime_error(input->name() + " has unknown dimension");

  if (numElements(input) == 0)
    throw std::runtime_error(input->name() + " is a zero-sized input");
```